packages/bsky/src/hydration/feed.ts

```diff
@@ -203,6 +206,57 @@ export class FeedHydrator {
     }, new HydrationMap<Threadgate>())
   }
 
+  async getThreadGateRecords(
+    uris: string[],
+    includeTakedowns = false,
+  ): Promise<Threadgates> {
+    const res = await this.dataplane.getThreadGateRecords({ uris })
+    return uris.reduce((acc, uri, i) => {
+      const record = parseRecord<ThreadgateRecord>(
+        res.records[i],
+        includeTakedowns,
+      )
+      return acc.set(uri, record ?? null)
+    }, new HydrationMap<Threadgate>())
+  }
+
+  async getPostGatesForPosts(
+    postUris: string[],
+    includeTakedowns = false,
+  ): Promise<PostGates> {
+    if (!postUris.length) return new HydrationMap<PostGate>()
+    const uris = postUris.map((uri) => {
+      const parsed = new AtUri(uri)
+      return AtUri.make(
+        parsed.hostname,
+        ids.AppBskyFeedPostgate,
+        parsed.rkey,
+      ).toString()
+    })
+    const res = await this.dataplane.getPostGateRecords({ uris })
```

Comment:
looks like this is the same logic as `getPostGateRecords` maybe we just call that method?

Reply:
same for `getThreadGateRecords` above 👍

Reply:
Ah yeah good call